filebeat/module/nginx/ingress_controller/ingest/pipeline.yml

```diff
@@ -139,18 +140,42 @@ processors:
           for (def item : ctx.nginx.ingress_controller.upstream_address_list) {
             last_upstream = item;
           }
-          StringTokenizer tok = new StringTokenizer(last_upstream, ":");
-          if (tok.countTokens()>1) {
-            ctx.nginx.ingress_controller.upstream.ip = tok.nextToken();
-            ctx.nginx.ingress_controller.upstream.port = Integer.parseInt(tok.nextToken());
-          } else {
-            ctx.nginx.ingress_controller.upstream.ip = last_upstream;
-          }
+
+          ctx.nginx.ingress_controller.upstream.address = last_upstream;
         }
         catch (Exception e) {
-          ctx.nginx.ingress_controller.upstream.ip = null;
-          ctx.nginx.ingress_controller.upstream.port = null;
+          ctx.nginx.ingress_controller.upstream.address = null;
         }
+  - grok:
+      field: nginx.ingress_controller.upstream.address
+      patterns:
+        - "^%{IPV4:nginx.ingress_controller.upstream.ip}:%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^\\[%{IPV6:nginx.ingress_controller.upstream.ip}\\]:%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^%{IPV6NOCOMPRESS:nginx.ingress_controller.upstream.ip}:%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^%{IPV6:nginx.ingress_controller.upstream.ip}%{IPV6PORTSEP}%{NUMBER:nginx.ingress_controller.upstream.port}$"
+        - "^%{IPV6:nginx.ingress_controller.upstream.ip}%{IPV6PORTSEP}%{POSINT:nginx.ingress_controller.upstream.port}$"
+      pattern_definitions:
+        IPV6NOCOMPRESS: '([0-9A-Fa-f]{1,4}:){7}[0-9A-Fa-f]{1,4}'
+        IPV6PORTSEP: '(?: port |[p#.])'
+      ignore_missing: true
+      ignore_failure: true
```

Comment:
So wondering if this has to be ignore_failure: false ?

Reply:
just to be on the safe side